pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -253,4 +259,10 @@ private static boolean isFitForNonScanBasedPlan(AggregationFunction[] aggregatio
     }
     return true;
   }
+
+  private static boolean canOptimizeFilteredCount(BaseFilterOperator filterOperator,
+      AggregationFunction[] aggregationFunctions) {
+    return filterOperator.canOptimizeCount() && (aggregationFunctions.length == 1
+        && aggregationFunctions[0].getType() == COUNT);
```

Comment:
Suggest changing the order of the checks because `canOptimizeCount()` can be more costly
```suggestion
    return aggregationFunctions.length == 1 && aggregationFunctions[0].getType() == COUNT && filterOperator.canOptimizeCount();
```